lib/commons/dom/find-up.js

```diff
@@ -9,33 +9,24 @@
  * @return {HTMLElement|null} Either the matching HTMLElement or `null` if there was no match
  */
 dom.findUp = function (element, target) {
-	'use strict';
```

Comment:
The problem is that if the child is a directly on the shadow root, it would start looking from the host (outside the shadow), even though the matches array still only contained elements inside the shadow tree.

Followup thought, should this use element.matches() instead of doing querySelectAll() and see if the element is included? I imagine that'd be faster.